server/src/Dao/RevisionDao.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace App\Dao;
+
+use Doctrine\DBAL\Connection;
+
+class RevisionDao extends AbstractDao {
+
+	/**
+	 * @param string[] $users
+	 * @param null $startDate
+	 * @param null $endDate
+	 * @param int $limit
+	 * @param null $continue
+	 * @return array
+	 */
+	public function getUserRevisionsInCommonPages(
+		$users, $startDate = null, $endDate = null, $limit = 50, $continue = null
+	) {
+		// build subquery to find common pages between users
+		$subQuery = $this->conn->createQueryBuilder();
+		$subQuery->select( 'rev_page' )
+			->from( 'revision_userindex' )
+			->where( 'rev_user_text in (:users)' )
+			->groupBy( 'rev_page' )
+			->having( 'count(distinct rev_user) > 1' );
+
+		// build query to find interaction between users in common pages
+		$query = $this->conn->createQueryBuilder();
+		$query->select( 'rev_id' )
+			->from( 'revision_userindex', 'r' )
+			->where( 'rev_user_text in (:users)' )
+			->andWhere( $query->expr()->in( 'rev_page', $subQuery->getSQL() ) )
+			->orderBy( 'rev_timestamp', 'asc' )
```

Comment:
I don't think this `orderBy` is being used because of the pagination.

Reply:
Pagination is not ordering, it only restrict results to >= rev_id. We need this

Reply:
uhhh.. I could be wrong on this.. but I think for token pagination to work properly, the results have to be ordered by the same thing you're keying off of. Which means this should be ordered by `rev_id` not by timestamp (although, they should always be the same order, I guess it could be off since timestamp is not unique).

Reply:
You are not, this is an oversight on my end. It just happened to work. I'll fix it